### DISCUSSION TITLE
Tapo C100 does not work / Tapo C210 works

### DISCUSSION THREAD

**Discussion by teuteuguy** [koush/scrypted#1202] [Q&A]
I just installed scrypted. 
Installed the different plugins. Both cameras C100 and C210 are configured inside the Tapo app.
When adding the C210 camera, scrypted works, I can view the stream no problem.
When adding the C100 camera, scrypted adds it, but cannot view the stream.

Both cameras are setup exactly the same way with the same rtsp account user/password.

Looking at the console for the C100:
...
[Rebroadcast Plugin]: rtsp outgoing message
[Rebroadcast Plugin]:  TEARDOWN rtsp://192.168.68.193:554/stream1 RTSP/1.0
[Rebroadcast Plugin]: CSeq: 3
[Rebroadcast Plugin]: User-Agent: Scrypted
[Rebroadcast Plugin]: Authorization: Basic dGV1dGV1Z3V5OlRlbXB0ZW1wMTIz
[Rebroadcast Plugin]: 
[Rebroadcast Plugin]: mainStream prebuffer session ended with error Error: auth failed
[Rebroadcast Plugin]:     at t.RtspClient.request (/@scrypted/prebuffer-mixin/main.nodejs.js:1:92275)
[Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]:     at async t.startRtspSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:144502)
[Rebroadcast Plugin]:     at async V.startPrebufferSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:117990)
[Rebroadcast Plugin]: Tapo C100 restarting prebuffer session in 5 seconds
[Rebroadcast Plugin]: rebroadcast error Error: rtsp socket closed
[Rebroadcast Plugin]:     at Socket.<anonymous> (/@scrypted/prebuffer-mixin/main.nodejs.js:1:144316)
[Rebroadcast Plugin]:     at Socket.emit (node:events:514:28)
[Rebroadcast Plugin]:     at Socket.emit (node:domain:489:12)
[Rebroadcast Plugin]:     at TCP.<anonymous> (node:net:323:12)

Seems theres' an authentication issue.
However it's the same auth.

Finally, testing the rtsp stream using vlc, same user/passowrd, works fine.

**Comment by teuteuguy** [comment]
Hi. Curious why this was moved to discussion ?

How can this be resolved ?
Happy to help where I can

**Reply by stncttr908** [reply]
If you’re not using the ONVIF plugin try that. That’s what I am using for my C100 and it works very well.

**Comment by koush** [comment]
probably need to specify port 2020 or something

**Comment by heinsenberg82** [comment]
I have the same problem, but my camera is a Tapo c210. The RTSP link works normally everywhere (VLC, iSpy, etc.), except in Scrypted. When I try to add the camera through the OVINF or RTSP plugin, it keeps producing authentication errors.

```
[Rebroadcast Plugin]: Tapo C210 restarting prebuffer session in 5 seconds
[Rebroadcast Plugin]: mainStream prebuffer session ended with error Error: auth failed
[Rebroadcast Plugin]:     at t.RtspClient.request (/server/volume/plugins/@scrypted/common/src/rtsp-server.ts:728:23)
[Rebroadcast Plugin]:     at process.response (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]:     at t.RtspClient.sdp [as describe] (/server/volume/plugins/@scrypted/common/src/rtsp-server.ts:763:15)
[Rebroadcast Plugin]:     at session (/server/volume/plugins/@scrypted/prebuffer-mixin/zip/src/rtsp-session.ts:66:13)
[Rebroadcast Plugin]:     at V.startPrebufferSession (/server/volume/plugins/@scrypted/prebuffer-mixin/zip/src/main.ts:588:9)

```
